core/templates/components/score-ring/score-ring.component.ts

```diff
@@ -66,14 +70,23 @@ export class ScoreRingComponent implements OnInit, OnChanges {
     }
   }
 
-  ngOnInit(): void {
-    this.circle = (
-      document.querySelector('.score-ring-circle') as SVGCircleElement);
+  ngAfterViewInit(): void {
+    this.circle = this.scoreRingElement.nativeElement;
     this.radius = this.circle.r.baseVal.value;
     this.circumference = (this.radius * 2 * Math.PI);
+
     this.circle.style.strokeDasharray = (
       `${this.circumference} ${this.circumference}`);
     this.circle.style.strokeDashoffset = this.circumference.toString();
+    // A reflow needs to be triggered so that the browser picks up the
+    // assigned starting values of stroke-dashoffset and stroke-dasharray before
+    // setting the transition-duration value to '5s', else the ring can be seen
+    // animating down to the starting position.
+    // clientHeight property triggers a reflow.
+
+    this.circle.clientHeight;
```

Comment:
Do we need this?

Reply:
By default, the stroke-dashoffset value of an SVG element is 0. Visually, this renders the score-ring as filled 100%. However, we need the score-ring to render as filled 0% upon initialization, and then fill up to the required score value that the learner has achieved.

To set the score-ring down to 0%, its dashoffset value is set to the ring's circumference.

The ring should be set to 0% immediately, and then animate back up to the required value. So when setting the dashoffset to the ring's circumference, the transition duration remains 0 (i.e. the default value). After the dashoffset has been set and the ring has been initialized, the transition duration is set to 5s, so subsequent animations can play out normally.

From what I've read, the problem with this is that even though dashoffset is set before changing the transition-duration to 5s, it can still be seen animating down to 0%, because the browser hasn't recalculated the position of the stroke yet.

To force the browser to recalculate the position of the stroke line before transitionDuration becomes 5s, a reflow needs to be triggered (so the browser recalculates the positions of all the DOM elements, including the stroke line). clientHeight is one of the properties that trigger a reflow.

Reply:
Oh I see. Thanks so much for that explanation! Let's remove the new line after the comment.

Reply:
> Let's remove the new line after the comment.

Done!